src/sidebar/components/test/tag-editor-test.js

```diff
@@ -1,63 +1,257 @@
+      wrapper.find('input').instance().value = '';
+
+      wrapper.find('input').simulate('keyup'); // simulates a selection
+      addTagsFail();
+    });
+
+    it('should not add a tag if the input is blank space', () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = '  ';
+      wrapper.find('input').simulate('keyup'); // simulates a selection
+      addTagsFail();
+    });
+
+    it('should not add a tag if its a duplicate of one already in the list', () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = 'tag1';
+      wrapper.find('input').simulate('keyup'); // simulates a selection
+      addTagsFail();
+    });
 
-    beforeEach(function() {
-      onEditTags = sinon.stub();
-      element = util.createDirective(document, 'tag-editor', {
-        onEditTags: { args: ['tags'], callback: onEditTags },
-        tags: ['foo'],
-      });
-      element.ctrl.onTagsChanged();
+    it('adds a tag via keypress `Enter`', () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = 'tag3';
+
+      wrapper.find('input').simulate('keypress', { key: 'Enter' });
+      addTagsSuccess(wrapper, ['tag1', 'tag2', 'tag3']);
     });
 
-    it('calls onEditTags handler', function() {
-      assert.calledWith(onEditTags, sinon.match(['foo']));
+    it('adds a tag via keypress `,`', () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = 'tag3';
+
+      wrapper.find('input').simulate('keypress', { key: ',' });
+      addTagsSuccess(wrapper, ['tag1', 'tag2', 'tag3']);
     });
 
-    it('saves tags to the store', function() {
-      assert.calledWith(fakeTags.store, sinon.match([{ text: 'foo' }]));
+    it('does not add a tag when key is not `,` or  `Enter`', () => {
+      const wrapper = createComponent();
+      wrapper.find('input').instance().value = 'tag3';
+
+      wrapper.find('input').simulate('keypress', { key: 'e' });
+      addTagsFail();
     });
   });
 
-  describe('#autocomplete', function() {
-    it('suggests tags using the `tags` service', function() {
-      const element = util.createDirective(document, 'tag-editor', {
-        tags: [],
-      });
-      element.ctrl.autocomplete('query');
-      assert.calledWith(fakeTags.filter, 'query');
+  context('when removing tags', () => {
+    /**
+     * Helper function to assert that a tag was correctly added
+     */
+    const removeTagsSuccess = tagList => {
+      // saves the suggested tags to the service
+      assert.isTrue(
+        fakeTagsService.store.calledWith(tagList.map(tag => ({ text: tag })))
+      );
+      // called the onEditTags callback prop
+      assert.isTrue(fakeOnEditTags.calledWith({ tags: tagList }));
+    };
+
+    it('removes `tag1` when clicking its delete button', () => {
+      const wrapper = createComponent();
+      assert.equal(wrapper.find('.tag-editor__edit').length, 2);
```

Comment:
This might warrant a comment that there are two of these elements because of the `fakeTagList` value

Reply:
I agree